sql/src/main/java/io/crate/metadata/settings/session/SessionSetting.java

```diff
@@ -29,25 +29,38 @@
 import io.crate.sql.tree.Expression;
 
 import java.util.List;
+import java.util.Map;
 import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 public class SessionSetting<T> {
 
+    public enum Property {
+        DESCRIPTION,
+        TYPE
+    }
+
     private final Consumer<Object[]> validator;
     private final Function<Object[], T> converter;
     private final BiConsumer<SessionContext, T> setter;
     private final Function<SessionSettings, String> getter;
+    private final Supplier<String> defaultValue;
+    private final Map<Property, String> properties;
 
     SessionSetting(Consumer<Object[]> validator,
                    Function<Object[], T> converter,
                    BiConsumer<SessionContext, T> setter,
-                   Function<SessionSettings, String> getter) {
+                   Function<SessionSettings, String> getter,
```

Comment:
not sure about this extension of SessionSetting, maybe we should make getter and defaultValue generic, but we do not really need a type of the setting that we return, so probably String is ok..

Reply:
I guess string is okay - since we expose it in `pg_settings` and we can't have mixed types there as result